client/state/data-layer/wpcom/sites/automated-transfer/initiate/test/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import {
+	initiateTransfer,
+	receiveResponse,
+	receiveError,
+	updateUploadProgress,
+} from '../';
+import {
+	getAutomatedTransferStatus,
+} from 'state/automated-transfer/actions';
+import {
+	pluginUploadError,
+	updatePluginUploadProgress,
+} from 'state/plugins/upload/actions';
+
+const siteId = 1916284;
+
+const ERROR_RESPONSE = {
+	error: 'invalid_input',
+	message: 'Invalid file type.',
+};
+
+describe( 'initiateTransfer', () => {
+	it( 'should dispatch an http request', () => {
+		const dispatch = sinon.spy();
+		initiateTransfer( { dispatch }, { siteId, pluginZip: 'foo' } );
+		expect( dispatch ).to.have.been.calledWithMatch( {
```

Comment:
Do you think it is also worth to test the number of `dispatch` calls with `callCount` or `calledOnce`? I've seen that done here and there.

Reply:
I think that would be getting into testing the data-layer framework in this case. I do think `callCount` and friends are useful for multiple dispatches, though.